runtime/src/main/scala-2/mdoc/internal/document/Printing.scala

```diff
@@ -15,17 +17,8 @@ object Printing {
       .foreach(text => out.appendAll(text.getChars))
   }
 
-  def printOneLine[T](value: T, out: StringBuilder, width: Int) = {
-    val chunk = BlackWhite
-      .tokenize(value, width)
-      .map(_.getChars)
-      .filterNot(_.iterator.forall(_.isWhitespace))
-      .flatMap(_.iterator)
-      .filter {
-        case '\n' => false
-        case _ => true
-      }
-    out.appendAll(chunk)
-
+  def printOneLine(value: Str, out: StringBuilder, width: Int) = {
+    out.appendAll(value.toString().replace("\n", " ").replaceAll("\\s+", " ")).take(width + 1)
```

Comment:
Hm, I feel like this _should_ also collapse the possible spaces inside a value, e.g. as part of a string constant for example?

But I struggle to reproduce it in tests.

Reply:
Ok, I don't see a good way of doing it without changing how binding works in general..
That said I'm not deeply familiar with worksheets, will take some time to read it _eventually_

Reply:
We would need a nice intermediate representation of the values, which we can print later on. I almost got it to work with pprint trees, but then pprint uses iterators and I could write things only once. So in the end we would need to reuse pprint trees by either inlining and changing them or by doing a fix upstream. That said, for Scala 2 we don't use pprint, just toString and I am afraid to change it since Martin was mad at us for not doing toString. 

All that makes me think that making it work proper currently is not really worth the effort and current approach is good enough :sweat_smile: